assembly/src/lib.rs

```diff
@@ -349,15 +358,19 @@ impl AsRef<str> for ModulePath {
 }
 
 impl Serializable for ModulePath {
-    fn write_into(&self, target: &mut ByteWriter) -> Result<(), SerializationError> {
-        target.write_str(&self.path)
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        target.write_u64(self.path.len() as u64);
+        target.write_bytes(self.path.as_bytes());
     }
```

Comment:
I think `u64` is too much - let's use `u16` and add TODOs to enforce this restriction via the constructor.